R/R/response.R

```diff
@@ -245,24 +246,46 @@ robyn_response <- function(InputCollect = NULL,
     metric_value_updated <- metric_value
     metric_ds_val <- NULL
   } else {
-    if ("last_date" %in% metric_ds) {
-      metric_ds <- tail(get_ds, 1)
-      metric_ds_loc <- which(get_ds == metric_ds)
+    if (grepl("last", metric_ds[1])) {
+      last_n <- ifelse(grepl("_", metric_ds[1]), as.integer(gsub("last_", "", metric_ds)), 1)
+      metric_ds <- tail(get_ds, last_n)
+      metric_ds_loc <- which(get_ds %in% metric_ds)
       metric_ds_val <- get_ds[metric_ds_loc]
-      if (!quiet) message("Using the last ds '", metric_ds_val, "' for the adstocked input value")
-    } else if (is.Date(as.Date(metric_ds, origin = "1970-01-01")) & length(metric_ds) == 1) {
-      metric_ds <- as.Date(metric_ds, origin = "1970-01-01")
-      metric_ds_loc <- which.min(abs(metric_ds - get_ds))
-      # Set closest date available if not exact date provided
-      if (!metric_ds %in% get_ds) {
-        metric_ds_val <- get_ds[which(abs(get_ds - metric_ds) == min(abs(get_ds - metric_ds)))]
-        if (!quiet) warning("Input 'metric_ds' (", metric_ds, ") was not valid. Picking closest date: ", metric_ds_val)
-      }
-      if (!quiet) message("Using ds '", metric_ds_val, "' for the adstocked input value")
+      if (!quiet) message("Using the last ", last_n, " ds (", v2t(metric_ds_val), ") for the adstocked input value")
     } else {
-      ## for length(metric_ds) > 1
+      if (all(is.Date(as.Date(metric_ds, origin = "1970-01-01")))) {
+        metric_ds <- as.Date(metric_ds, origin = "1970-01-01")
+        if (length(metric_ds) == 1) {
+          metric_ds_loc <- which.min(abs(metric_ds - get_ds))
+          # Set closest date available if not exact date provided
+          if (!metric_ds %in% get_ds) {
+            metric_ds_val <- get_ds[which(abs(get_ds - metric_ds) == min(abs(get_ds - metric_ds)))]
+            if (!quiet) warning("Input 'metric_ds' (", metric_ds, ") was not valid. Picking closest date: ", metric_ds_val)
+          } else {
+            metric_ds_val <- metric_ds
+          }
+          if (!quiet) message("Using ds '", metric_ds_val, "' for the adstocked input value")
+        } else {
+          ## Multiple metric_ds (and metric_value)
```

Comment:
this is the case when metric_ds has multiple value and metric_value has only 1, e.g. user inputs  "from-to" dates, and we just divide metric_value by the length of the from-to dates. this is the case when user has a budget for x days, but don't know how to split the spend for each day.

we can also allow users to input all desired periods, like for 3 days, 2023-01-01, 2023-01-02, 2023-01-03, while also inputting 3 metric_value like c(200, 150, 250). This is the case when the user has a future media plan and know when they want to spend how much. 

I agree that we can think of a more intuitive argument name than 'metric_ds'. we can consider it along the way.

Reply:
I think you missed the part of the code where this is already implemented haha. I've enabled the "last" or "last_n" inputs for "metric_ds" where n can be any integer value to pick from the latest n available days. The multiple metric_value is also enabled already and divided by total metric_ds dates. I still have to add "all" as a possible input though.

Reply:
Input "all" enabled. Please @gufengzhou can you check the outputs and see if they are as expected, especially for +1 ds. You can use these:
```
devtools::load_all()
robyn_response(
  InputCollect = InputCollect,
  OutputCollect = OutputCollect,
  select_model = select_model,
  media_metric = select_media,
  metric_value = metric_value,
  metric_ds = "last",
  quiet = FALSE
)
robyn_response(
  InputCollect = InputCollect,
  OutputCollect = OutputCollect,
  select_model = select_model,
  media_metric = select_media,
  metric_value = metric_value,
  metric_ds = "last_3",
  quiet = FALSE
)
robyn_response(
  InputCollect = InputCollect,
  OutputCollect = OutputCollect,
  select_model = select_model,
  media_metric = select_media,
  metric_value = metric_value,
  metric_ds = "all",
  quiet = FALSE
)
robyn_response(
  InputCollect = InputCollect,
  OutputCollect = OutputCollect,
  select_model = select_model,
  media_metric = select_media,
  metric_value = metric_value,
  metric_ds = c("2019-01-01", "2019-11-11"),
  quiet = FALSE
)
```